File: cs50_tasks/week2_Arrays/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// try to readability test is the Coleman-Liau index.
//The Coleman-Liau index of a text is designed to output what (U.S.)
//grade level is needed to understand the text. The formula is:

//index = 0.0588 * L - 0.296 * S - 15.8

//Here, L is the average number of letters per 100 words in the text,
//and S is the average number of sentences per 100 words in the text.

int let = 0, spases = 0, symbols = 0;
void getSymb(string text);
int getIndex(string text);
void printIndex(string text);

int main (void)
{
    string text = get_string("Text: ");
    printIndex(text);
}

void printIndex(string text)
{
    int index = getIndex(text);
    if(index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n" , index);
    }
}

int getIndex(string text)
{
    float index;
    getSymb(text);
    float inL, inS, ind;
    ind = 100.00 / (spases + 1);
    inL = let * ind;
    inS = symbols * ind;
    index = 0.0588 * inL - 0.296 * inS - 15.8;


    return round(index);
}

void getSymb(string text)
{
    int n = strlen(text);
    for(int i = 0; i < n; i++)
    {
        char c = text[i];
        if(isalpha(c)) let++;
        if(isspace(c)) spases++;
        if(c == '.' || c == '!' || c == '?') symbols++;
    }
}

File: cs50_tasks/week6_Python/pset6/readability/readability.py
from cs50 import get_string
import re


def main():
    # get text
    text = get_string("Text: ")
    # let magic begin
    print_index(text)


# find index
def get_index(text):
    # words_index?
    words_index = 100.00/len(re.split(" ", text))
    # quantity of sentences in 100 words
    sentesces_index = len(re.findall("[!\.\?]", text)) * words_index
    # quantity of letters
    letter_index = len(re.findall("[a-zA-Z]", text)) * words_index
    # return index
    index = 0.0588 * letter_index - 0.296 * sentesces_index - 15.8
    return round(index)

# print index


def print_index(text):
    # get index
    index = get_index(text)
    # print result
    if index < 1:
        print("Before Grade 1n")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


main()